samples/Jira.gs

```diff
@@ -0,0 +1,113 @@
+var CLIENT_SECRET = '...';
+
+// The key to use when storing the cloudid.
+var CLOUDID_KEY = 'cloudid';
+
+/**
+ * Authorizes and makes a request to the UltraCart API.
+ */
+function run() {
+  var service = getService();
+  if (service.hasAccess()) {
+    var cloudid = getCloudId(service);
+    var url = 'https://api.atlassian.com/ex/jira/' + cloudid +
+        '/rest/api/3/myself';
+    var response = UrlFetchApp.fetch(url, {
+      headers: {
+        Accept: 'application/json',
+        Authorization: 'Bearer ' + service.getAccessToken()
+      }
+    });
+    var result = JSON.parse(response.getContentText());
+    Logger.log(JSON.stringify(result, null, 2));
+  } else {
+    var authorizationUrl = service.getAuthorizationUrl();
+    Logger.log('Open the following URL and re-run the script: %s',
+        authorizationUrl);
+  }
+}
+
+/**
+ * Gets the cloudid of the Jira site to operate against. This implementation
+ * selects the first authorized site, but in a real application you'd probably
+ * want to let the user select which site to operate against. After the first
+ * run the cloudid is saved into the service's storage layer for easy retrieval.
+ * @param {OAuth.Service_} service The authorized service.
+ * @returns {string} The cloudid of the site to operate on.
+ */
+function getCloudId(service) {
+  var cloudid = service.getStorage().getValue(CLOUDID_KEY);
+  if (cloudid) return cloudid;
+  // Get the cloudid of the first site the user has access to.
+  var url = 'https://api.atlassian.com/oauth/token/accessible-resources';
+  var response = UrlFetchApp.fetch(url, {
+    headers: {
+      Accept: 'application/json',
+      Authorization: 'Bearer ' + service.getAccessToken()
+    }
+  });
+  var result = JSON.parse(response.getContentText());
+  cloudid = result[0].id;
+  service.getStorage().setValue(CLOUDID_KEY, cloudid);
+  return cloudid;
+}
+
+/**
+ * Reset the authorization state, so that it can be re-tested.
+ */
+function reset() {
+  getService().reset();
+}
+
+/**
+ * Configures the service.
+ */
+function getService() {
+  return OAuth2.createService('Jira')
+      // Set the endpoint URLs.
+      .setAuthorizationBaseUrl('https://accounts.atlassian.com/authorize')
+      .setTokenUrl('https://accounts.atlassian.com/oauth/token')
+
+      // Set the client ID and secret.
+      .setClientId(CLIENT_ID)
+      .setClientSecret(CLIENT_SECRET)
+
+      // Set the name of the callback function that should be invoked to
+      // complete the OAuth flow.
+      .setCallbackFunction('authCallback')
+
+      // Set the property store where authorized tokens should be persisted.
+      .setPropertyStore(PropertiesService.getUserProperties())
+
+      // Set the scope and other paramaeters required by Atlassian.
+      .setScope('read:jira-user')
+      .setParam('audience', 'api.atlassian.com')
+      .setParam('prompt', 'consent');
+}
+
+/**
+ * Handles the OAuth callback.
+ */
+function authCallback(request) {
```

Comment:
I'd personally prefer:

```js
function authCallback(request) {
  var output = getService().handleCallback(request) ? 'Success!' : 'Denied';
  return HtmlService.createHtmlOutput(output);
}
```

But I can understand if you're trying to keep this sample consistent with other samples.

Reply:
That and I try to avoid ternary operators when possible, since I think they aren't always the most readable.